bool left
bool right
bool up
bool down
